### PR TITLE
Fix Dockerfile so Auto-GPT can be built in Docker.

### PR DESCRIPTION
**Pull request by ksylvan**
The original Dockerfile is based on an older Python image and does not build:

```text
------                                                                                                                                                                                                                                                               
 > [4/4] RUN pip install -r requirements.txt:                                                                                                                                                                                                                        
#0 0.841 ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'                                                                                                                                                            
#0 2.018 
#0 2.018 [notice] A new release of pip available: 22.3.1 -> 23.0.1
#0 2.018 [notice] To update, run: pip install --upgrade pip
------
Dockerfile:6
```

The updated version is latest Python image, based on alpine3.17:

```text
% docker build -t test .
[+] Building 33.1s (9/9)                                                                                                                                                                                                                                             
[+] Building 33.1s (9/9) FINISHED
[...]

% docker run --rm -it test /bin/sh
/app # python main.py
playsound is relying on another python subprocess. Please use `pip install pygobject` if you want playsound to run more efficiently.
Welcome to Auto-GPT!  Enter the name of your AI and its role below. Entering nothing will load defaults.
Name your AI:  For example, 'Entrepreneur-GPT'
AI Name: 
```

### CODE REVIEW

**Review by HusseinMansourMohd — DISMISSED** [review]
works

**Comment by ksylvan** [comment]
@HusseinMansourMohd Can you re-review this? Thanks.

File: Dockerfile
Comment (kreneskyp) [inline Dockerfile]: I suggest sticking with `python:3.11` or `python:3.11-slim` for a size optimized image. Some science libraries don't work on alpine unfortunately. It has a different c compiler.

Comment (Torantulino) [reply Dockerfile]:
```suggestion
FROM python:3.11
```

File: README.md
Comment (Torantulino) [inline README.md]:
```suggestion
5. Rename `.env.template` to `.env` and fill in your `OPENAI_API_KEY`. If you plan to use Speech Mode, fill in your `ELEVEN_LABS_API_KEY` as well.
```

File: README.md
Comment (Torantulino) [inline README.md]:
```suggestion
5. Rename `.env.template` to `.env` and fill in your `OPENAI_API_KEY` and `PINECONE_API_KEY`. If you want to use a higher quality Speech Mode, fill in your `ELEVEN_LABS_API_KEY` as well.
```

File: Dockerfile
Comment (ksylvan) [reply Dockerfile]: Cool. Thanks.

File: README.md
Comment (ksylvan) [reply README.md]: Why rename? If you rename it, then `git status` shows a missing file, but copying it and changing it works correctly, since `.env` is in the `.gitignore` file.

**Review by ksylvan — COMMENTED** [review]
LGTM, but I have the one question about "copy" versus "rename" of the .env template file.

**Comment by ksylvan** [comment]
Updated the PR @Torantulino - though I believe that the README should encourage people to copy the `.env.template` to `.env`, not *rename* it. The difference being that if you rename it and modify it, you see this in git:

```bash
 Auto-GPT % git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        deleted:    .env.template
```

But if you copy it and modify it, you have a clean repo.

**Comment by aosan** [comment]
> Updated the PR @Torantulino - though I believe that the README should encourage people to copy the `.env.template` to `.env`, not _rename_ it. The difference being that if you rename it and modify it, you see this in git:
> 
> ```shell
>  Auto-GPT % git status
> On branch master
> Your branch is up to date with 'origin/master'.
> 
> Changes not staged for commit:
>   (use "git add/rm <file>..." to update what will be committed)
>   (use "git restore <file>..." to discard changes in working directory)
>         deleted:    .env.template
> ```
> 
> But if you copy it and modify it, you have a clean repo.

great insight :+1: 
.env.template is also a reference for any upcoming configuration values. We recently had pinecone added, a Redis PR has been done by @BillSchumacher
A `diff .env .env.template` would be a quick way to see the configuration delta.

**Comment by ksylvan** [comment]
@aosan says:
> A diff .env .env.template would be a quick way to see the configuration delta.

Yes. Exactly. Which is why I changed it to "copy" in my update of the README.md that also adds the running in docker instructions.

**Comment by ksylvan** [comment]
@Torantulino Can you please review this again? Thank you.

**Comment by ksylvan** [comment]
> still not building

@riccitensor Are you sure your Dockerfile has the changes from this PR?

The Dockerfile should look like this:

```Dockerfile

FROM python:3.11

WORKDIR /app

COPY . /app/

RUN pip install -r requirements.txt

CMD ["python", "scripts/main.py"]
```

And docker build does this:

```text
[+] Building 29.4s (8/9)                                                                                                      
 => [internal] load build definition from Dockerfile                                                                     0.0s
 => => transferring dockerfile: 154B                                                                                     0.0s
 => [internal] load .dockerignore                                                                                        0.0s
 => => transferring context: 240B                                                                                        0.0s
 => [internal] load metadata for docker.io/library/python:3.11                                                           8.0s
 => [auth] library/python:pull token for registry-1.docker.io                                                            0.0s
 => [1/4] FROM docker.io/library/python:3.11@sha256:7adb2f6d6b0fdaf2d3029c42b5a40833589f969c18728f5b5b126a61394848b6     6.7s
 => => resolve docker.io/library/python:3.11@sha256:7adb2f6d6b0fdaf2d3029c42b5a40833589f969c18728f5b5b126a61394848b6     0.0s
 => => extracting sha256:8022b074731d9ecee7f4fba79b993920973811dda168bbc08636f18523b90122                                1.1s
 => => extracting sha256:7971239fe1d69763272ccc0b2527efa95547d37c53630ed0a71db4e00d3ef964                                0.1s
 => => extracting sha256:26c861b53509d61c37240d2f80efb3a351d2f1d7f4f8e8ec2e5004c1d86af89c                                0.1s
 => => extracting sha256:1714880ecc1c021a5f708f4369f91d3c2c53b998a56d563d0a9aa9be2488d794                                1.2s
 => => extracting sha256:895a945a1f9ba441c2748501c4d46569edfbc2bfbdb9b47d41e753e752247fdc                                3.3s
 => => extracting sha256:2a83fe9e3053a5f52e699f1078c2abeaad849bb81da9068f1aa541bb5673a21f                                0.3s
 => => extracting sha256:7a61191bcafc61c6dee2d4933ada6f4de7b1056931ab31dcb39a1bea43d2c420                                0.4s
 => => extracting sha256:533b6245815e096c6ea94acac9d2a4047401a045e0980e761b2228fb48a110ad                                0.0s
 => => extracting sha256:8fbc2ab8c37f3d5b1ed6dce6bf3197710992d189a0ee50edbce06e750b4b14d1                                0.1s
 => [internal] load build context                                                                                        0.0s
 => => transferring context: 88.99kB                                                                                     0.0s
 => [2/4] WORKDIR /app                                                                                                   0.4s
 => [3/4] COPY . /app/                                                                                                   0.0s
 => [4/4] RUN pip install -r requirements.txt                                                                           14.3s
 => => #   Downloading googleapis_common_protos-1.59.0-py2.py3-none-any.whl (223 kB)                                         
 ```

**Comment by ksylvan** [comment]
@Torantulino updated this PR to add Pillow to the requirements.txt (and sorted it) so that PR #416 works in Docker too.

**Comment by ksylvan** [comment]
@riccitensor Glad to hear. You can also run it from my dockerhub: https://hub.docker.com/r/kayvan/auto-gpt

```
mkdir ~/AGPT
docker run --rm -it -v $HOME/AGPT:/app/auto_gpt_workspace kayvan/auto-gpt
```

And the workspace files end up in `~/AGPT` with no `__pycache__` files in the repo directory.

**Review by nponeccop — CHANGES_REQUESTED** [review]
IMO it should be split into independent PRs:
- gitignore improvements
- README improvements
- requirements improvements
- Docker building improvements

It isn't a good idea to mix unrelated changes in a single PR, especially now that there are many PRs waiting for approval and merge.

**Comment by Torantulino** [comment]
Can you please submit this as a simple atomic PR, changing a single thing?
Your suggestions for the `README`, `.gitignore`, etc can be submitted as separate commits.

**Comment by ksylvan** [comment]
> Can you please submit this as a simple atomic PR, changing a single thing?
> Your suggestions for the `README`, `.gitignore`, etc can be submitted as separate commits.

Yes. Will do. Thank you @Torantulino

**Pull request closed without merging**
